### DISCUSSION TITLE
FaaS

### DISCUSSION THREAD

**Discussion by nutjob4life** [MrKinau/FishingBot#141] [Ideas]
What might be fun (and to serve as a small subsidiary income aside from the <kbd> ❤️Sponsor</kbd> button—which I did!) is to maybe offer fishing as a service or "FaaS".

I could envision a cloud app—say on EC2 or Fargate or Azure or whatever—where you'd spin up fishing bots on demand that could join servers or Realms and fish away. Maybe offer a little web control panel for each bot where you could monitor logs, turn them off and on, etc. It'd be nice if Mojang offered OAuth so users wouldn't have to share their credentials (but maybe that's a possibility with the transition to Microsoft accounts). Charge a little amount to keep the bot running plus an overhead fee which is profit.

Anyway, just a weird thought that passed through my head 😝

**Comment by MrKinau** [comment]
I already thought of a webservice for fishing, but there are some problems with it (as you already mentioned).
1. If users want to fish with their own account, they need to give credentials or a session token or the service have to use alt accounts (I'm not a fan of alt accounts).
2. I do not want to waste my time for sth. which won't get used. Maybe there are some weird people using the FishingBot currently, but I think it's just a niche and no one (or not that many people) are willing to pay to get some fish.
3. I need to manage all the financing stuff. Patreon is pretty simple and offers great ways for me to not worry about taxes, but a FaaS (great name btw) wouldn't be that easy.

So I think it's a nice idea and maybe someday I'll pick it up, but atm I don't plan to create a FaaS, but as the project is licensed under GPLv3 you're allowed to do so 😉
